### PR TITLE
HPC Benchmark implementation

### PR DESCRIPTION
**Pull request by gmtiddia**
Implementation of iaf_psc_alpha neuron model, together with example, test and documentation.
First implementation of the HPC Benchmark, with static weight connections.

Warning: there is still a difference with respect to NEST regarding the firing rates of the populations. An issue will be raised.

### CODE REVIEW

File: python/examples/plot_iaf_psc_alpha.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+conn_spec={"rule": "all_to_all"}
+
+syn_spec_ex={'receptor':0, 'weight': weight[0], 'delay': delay[0]}
+syn_spec_in={'receptor':1, 'weight': weight[1], 'delay': delay[1]}
+ngpu.Connect(spike, neuron, conn_spec, syn_spec_ex)
+ngpu.Connect(spike, neuron, conn_spec, syn_spec_in)
+
+record = ngpu.CreateRecord("", ["V_m_rel"], [neuron[0]], [0])
+#voltmeter = nest.Create('voltmeter')
```
Comment (jhnnsnk) [inline python/examples/plot_iaf_psc_alpha.py:27]: Remove commented lines.

File: python/test/nest_iaf_psc_alpha.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+weight = [1.0, -2.0]
+
+conn_spec={"rule": "all_to_all"}
+
+syn_spec_ex={'weight': weight[0], 'delay': delay[0]}
+syn_spec_in={'weight': weight[1], 'delay': delay[1]}
+nest.Connect(spike, neuron, conn_spec, syn_spec_ex)
+nest.Connect(spike, neuron, conn_spec, syn_spec_in)
+
+#record = nest.CreateRecord("", ["V_m_rel"], [neuron[0]], [0])
```
Comment (jhnnsnk) [inline python/test/nest_iaf_psc_alpha.py:26]: Here and below, shouldn't the commented lines be removed?

File: python/examples/plot_iaf_psc_alpha.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import sys
+import nestgpu as ngpu
+import numpy as np
+tolerance = 0.00005
```
Comment (suku248) [inline python/examples/plot_iaf_psc_alpha.py:4]: `tolerance` is not used

File: python/examples/plot_iaf_psc_alpha.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
```
Comment (suku248) [inline python/examples/plot_iaf_psc_alpha.py:5]: A description of the purpose of this example would be useful.

File: python/examples/temp_mpi.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
+import sys
+import ctypes
+import nestgpu as ngpu
+from random import randrange
+
```
Comment (suku248) [inline python/examples/temp_mpi.py:5]: A description of the purpose of this example would be useful.

File: python/examples/plot_iaf_psc_alpha.py (hunk 1 above)
Comment (gmtiddia) [reply python/examples/plot_iaf_psc_alpha.py:27]: Done

File: python/examples/plot_iaf_psc_alpha.py (hunk 4 above)
Comment (gmtiddia) [reply python/examples/plot_iaf_psc_alpha.py:14]: Right, I've just removed it.

File: python/examples/temp_mpi.py (hunk 5 above)
Comment (gmtiddia) [reply python/examples/temp_mpi.py:5]: Description added.

**Review by jhnnsnk — APPROVED** [review]
Thank you!

File: python/hpc_benchmark/hpc_benchmark.py [hunk 6]
```diff
@@ -0,0 +1,620 @@
...
+
+import os
+import sys
+import json
+import numpy as np
+import scipy.special as sp
+import matplotlib.pyplot as plt
+
+from time import perf_counter_ns
+from mpi4py import MPI
```
[the remark this answers is not in the document]
Comment (gmtiddia) [reply python/hpc_benchmark/hpc_benchmark.py]: Now replaced

File: python/hpc_benchmark/hpc_benchmark.py [hunk 7]
```diff
@@ -0,0 +1,620 @@
...
+parser.add_argument("--path", type=str, default=".")
+parser.add_argument("--seed", type=int, default=12345)
+args = parser.parse_args()
+
+M_INFO = 10
+M_ERROR = 30
+
+ngpu.ConnectMpiInit()
+
+comm = MPI.COMM_WORLD
```
[the remark this answers is not in the document]
Comment (gmtiddia) [reply python/hpc_benchmark/hpc_benchmark.py]: Yes, now retrieved from NEST GPU kernel.

**Pull request merged**